examples/training_flan-t5-xl.ipynb

```diff
@@ -0,0 +1,719 @@
+    "\n",
+    "- [__flan-t5-xl__](https://huggingface.co/google/flan-t5-xl/tree/main) is a large pre-trained transformer with 3 billion parameters, about 10 times larger than BERT-large\n",
+    "- [__tensor_parallel__](https://github.com/BlackSamorez/tensor_parallel) is a library that splits your model between GPUs in 2 lines of code\n",
+    "\n",
+    "You can run this notebook on your own hardware or using __[kaggle's free cloud instances with dual T4](https://www.kaggle.com/product-feedback/361104)__ (requires phone verification).\n",
+    "\n",
+    "The code is based on [this tutorial](https://github.com/huggingface/transformers/tree/main/examples/pytorch/summarization), but supports way larger models. It was originally run on **four rusty 1080Ti**, just to prove that it's possible. If you're running on something more serious, you can probably tune batch size."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/jheuristic/anaconda3/envs/py38_petals_yozh/lib/python3.8/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    }
+   ],
+   "source": [
+    "import torch\n",
+    "import tensor_parallel as tp\n",
+    "import transformers\n",
+    "\n",
+    "tokenizer = transformers.AutoTokenizer.from_pretrained(\"google/flan-t5-xl\")\n",
+    "model = transformers.T5ForConditionalGeneration.from_pretrained(\n",
+    "    \"google/flan-t5-xl\", torch_dtype=torch.bfloat16, low_cpu_mem_usage=True, offload_state_dict=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Apply tensor_parallel\n",
+    "\n",
+    "You wrap the model - and it becomes parallel across GPUs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Using automatic config: no tensor parallel config provided and no predefined configs can be used\n",
+      "Using ZeRO-3 sharding for 249856 non tensor-parallel parameters\n"
+     ]
+    }
+   ],
+   "source": [
+    "model = tp.tensor_parallel(model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_ids = tokenizer(\"A cat sat on a mat\", return_tensors=\"pt\").input_ids.to(\"cuda\")\n",
+    "output_ids = tokenizer(\"A cat sat did not sit on a mat\", return_tensors=\"pt\").input_ids.to(\"cuda\")\n",
+    "\n",
+    "# forward and backward works as usual\n",
+    "loss = model(input_ids=input_ids, labels=output_ids).loss\n",
+    "loss.backward()  # check nvidia-smi for gpu memory usage :)"
```

Comment:
It would make more sense to include some simple function for fancy memory footprint printing

Reply:
We agreed to do it later